iml-mailbox/src/lib.rs

```diff
@@ -83,94 +102,87 @@ impl MailboxSenders {
+async fn get_task_by_name(
+    x: impl ToString,
+    pool: &iml_orm::DbPool,
+) -> Result<Option<Task>, iml_orm::tokio_diesel::AsyncError> {
+    Task::by_name(x).first_async(pool).await.optional()
+}
+
+/// Given an task name and `mpsc::UnboundedReceiver` handle,
 /// this fn will create or open an existing file in append mode.
 ///
 /// It will then write any incoming lines from the passed `mpsc::UnboundedReceiver`
 /// to that file.
 pub async fn ingest_data(
-    address: PathBuf,
+    task: String,
     mut rx: mpsc::UnboundedReceiver<Incoming>,
-) -> Result<(), std::io::Error> {
-    tracing::info!("Starting ingest for {:?}", address);
+) -> Result<(), MailboxError> {
+    tracing::info!("Starting ingest for {:?}", task);
 
-    let mut file = OpenOptions::new()
-        .append(true)
-        .create(true)
-        .open(address)
-        .await?;
+    let pool = iml_orm::pool()?;
+
+    let task = match get_task_by_name(&task, &pool).await? {
+        Some(t) => t,
+        None => {
+            tracing::info!("Task {} not found", task);
+
+            return Err(MailboxError::NotFound(format!("Failed to find {}", task)));
+        }
+    };
+
+    let mut count: i64 = 0;
 
     while let Some(incoming) = rx.next().await {
         match incoming {
-            Incoming::Line(mut line) => {
-                if !line.ends_with('\n') {
-                    line.extend(['\n'].iter());
+            Incoming::Line(line) => {
+                tracing::trace!("handling line {:?}", &line);
+
+                let mut map: HashMap<String, serde_json::Value> = serde_json::from_str(&line)?;
+
+                if let Some(fid) = map.remove("fid".into()) {
+                    let fid = LustreFid::from_str(fid.as_str().ok_or(MailboxError::NotFound(
+                        format!("Failed to find fid in {}", line),
+                    ))?)?;
+
+                    // Get "other" data if it exists
+                    let data = match map.iter().next() {
+                        Some((_, v)) => v.clone(),
+                        None => json!({}),
+                    };
+
+                    tracing::trace!("Inserting fid:{:?} data:{:?} task:{:?}", fid, data, task);
+                    insert_fidtask(fid, data, &task)
+                        .execute_async(&pool)
+                        .await?;
+
+                    count += 1;
+                } else {
+                    tracing::error!("No FID for task {} in line {:?}", &task.name, &line);
                 }
-
-                tracing::trace!("handling line {:?}", line);
-
-                file.write_all(line.as_bytes()).await?;
             }
             Incoming::EOF(tx) => {
-                file.flush().await?;
-
                 let _ = tx.send(());
             }
         }
     }
 
-    file.flush().await?;
+    task::increase_total(task.id, count)
+        .execute_async(&pool)
+        .await?;
 
     Ok(())
 }
-
-#[cfg(test)]
-mod tests {
```

Comment:
Why was the test deleted?

Reply:
because it now needs a database connection